src/main/java/org/spongepowered/common/mixin/core/world/MixinWorldServer.java

```diff
@@ -735,6 +738,18 @@ public void onUpdateTick(Block block, net.minecraft.world.World worldIn, BlockPo
 
     @Redirect(method = "addBlockEvent", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/WorldServer$ServerBlockEventList;add(Ljava/lang/Object;)Z", remap = false))
     public boolean onAddBlockEvent(WorldServer.ServerBlockEventList list, Object obj, BlockPos pos, Block blockIn, int eventId, int eventParam) {
+        // We fire a Pre event to make sure our captures do not get stuck in a loop.
+        // This is very common with pistons as they add block events while blocks are being notified.
+        if (blockIn instanceof BlockPistonBase) { // if it's a piston, try to simulate it
```

Comment:
This whole branching if else is just ugly. Simplify it down to a single if statement for lazy evaluation. 

```java
if ((blockIn instanceof BlockPistonBase && SpongeCommonEventFactory.handlePistonEvent(this, list, obj, pos, blocking, eventId, eventParam)
   || SpongeCommonEventFactory.callChangeBlockEventPre(this, pos, NamedCause.of(NamedCause.BLOCK_EVENT, this)).isCancelled()) {
  return false;
}
```